packages/canopee-react/src/prospect-client/List/List/ListCommon.tsx

```diff
@@ -0,0 +1,23 @@
+import { Children, ComponentProps, ComponentType, isValidElement } from "react";
+import {
+  CardCommon as Card,
+  type CardCommonProps,
+} from "../../Card/CardCommon";
+
+export type ListProps = CardCommonProps<"ul">;
+
+type ListPropsCommon = ListProps & {
+  CardComponent: ComponentType<ComponentProps<typeof Card>>;
+};
+
+export const ListCommon = ({
+  children,
+  CardComponent,
+  ...listProps
+}: ListPropsCommon) => (
+  <CardComponent as="ul" {...listProps}>
+    {Children.toArray(children).map(
+      (child) => isValidElement(child) && <li key={child.key}>{child}</li>,
+    )}
```

Comment:
Plus besoin du `isValidElement` car le `Children.toArray` fait le tri : https://react.dev/reference/react/Children#children-toarrayhttps://react.dev/reference/react/Children#children-toarray
(voir la partie `Caveats`)


```suggestion
    {Children.toArray(children).map(
      (child) => <li key={child.key}>{child}</li>,
    )}
```

Reply:
pas tout à fait, il enlève juste les noeuds vides (null, undefined et boolean), par contre si ton child est 25 ça passe alors qu'avec un isValidElement, ça ne passe pas
Et il y a d'autres exemples : 
<img width="509" height="141" alt="image" src="https://github.com/user-attachments/assets/b1997daa-2430-4f6c-99b0-f3989d5885c1" />
https://react.dev/reference/react/isValidElement

Reply:
Est-il vraiment nécessaire d'avoir besoin d'une vérification à ce niveau là alors que c'est juste un ReactNode à mettre dans un `li` ?

Reply:
c'est ainsi que c'était fait, ça me va si tu veux faire moins de vérifications, on peut demander l'avis d'autres dev : @pplancq @MartinWeb @johnmeunier

Reply:
par contre, j'ai testé et typescript râle car certains types éléments ne possèdent pas de key.

Reply:
pour moi le isValidElement n'est pas utile en effet, pour le child.key j'ai la solution dans le DS il y a un helper generateId qui est faite pour ca
exemple ici https://github.com/AxaFrance/design-system/blob/d7a5fc343a5f0838ae38880d151c7368076ebf24/packages/canopee-react/src/prospect-client/Form/FileUpload/FileUpload/FileUploadCommon.tsx#L58C24-L58C34

Reply:
done